packages/npm-is/README.md

```diff
@@ -0,0 +1,102 @@
+Detect the package manager client running the current script.
+Detects NPM and yarn, as well as any hard executable path.
+
+Use this when your project requires a specific package manager to work
+correctly, such as a monorepo using Yarn workspaces.
+It can help your package scripts exit early if they're being invoked outside of
+the required package manager.
+
+## Recommended Usage
+
+In a monorepo, or any repo with `private: true`, it's safe to catch things as
+early as possible.
+Add a `preinstall` script to prevent the wrong package manager from even
+beginning to install dependencies.
+Use `npx` to dynamically install `npm-is` before any other
+modules are installed.
+
+```diff
+ {
+   "name": "my-yarn-only-monorepo",
+   "version": "1.0.0",
+   "private": true,
+   "workspaces": [
+     "packages/*"
+   ]
+   "scripts": {
++    "preinstall": "npx npm-is yarn"
+   }
+ }
+```
+
+Now, this happens when you try to `npm install`.
+
+```sh
+$ npm install
+
+> my-yarn-only-monorepo@1.0.0 preinstall /Users/me/repo
+> npx npm-is yarn
+
+/Users/me/repo/node_modules/npm-is/npm-is.js:54
+    throw new WrongPackageManagerError(allowed, invoked);
+    ^
+
+Error: This project can only be used with the "yarn" package manager, but it was
+invoked by "npm", which is not supported.
+```
+
+## Install
+
+**Don't install this as a dependency if you're going to use it in the**
+**[recommended way described above](#Recommended_usage).**
+
+**Instead, run it with `npx` so that it works before dependencies are installed!**
```

Comment:
This does not make sense anymore because that is the problem at hand 😅 . Leaving it for now till I check with Zetlen.

Reply:
Instead how about changing the usage of `npx` to `yarn add -D @magento/npm-is && yarn @magento/npm-is run validate yarn` that way we are only installing the required package and checking for the package manager before installing the whole package.json file?

Reply:
I even tried `npm exec` and it fails with a similar error because the value of `process.env.npm_execpath` is the path to the NPM CLI.